specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2023-07-01-preview/appplatform.json

```diff
@@ -0,0 +1,15096 @@
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "AppPlatform"
+        ],
+        "description": "Delete the API portal custom domain.",
+        "operationId": "ApiPortalCustomDomains_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiPortalNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiPortalCustomDomainParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response indicates the API portal custom domain has deleted."
+          },
+          "202": {
+            "description": "Accepted. The response indicates the API portal custom domain is now deleting."
+          },
+          "204": {
+            "description": "Success. The response indicates the API portal custom domain doesn't exist or has deleted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "ApiPortalCustomDomains_Delete": {
+            "$ref": "./examples/ApiPortalCustomDomains_Delete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppPlatform/Spring/{serviceName}/apiPortals/{apiPortalName}/domains": {
+      "get": {
+        "tags": [
+          "AppPlatform"
+        ],
+        "description": "Handle requests to list all API portal custom domains.",
+        "operationId": "ApiPortalCustomDomains_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiPortalNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describes the list of API portal custom domains.",
+            "schema": {
+              "$ref": "#/definitions/ApiPortalCustomDomainResourceCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+     
```

Comment:
Consider modeling it as an enums if the property has a finite set of values?
refs: https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi015-avoid-modelling-a-property-as-string-type-when-it-should-be-an-enums

Reply:
@LGDoor Can you take a look for the comment for `languageFramework` field?

Reply:
Just tried but it results in an `EnumMustNotHaveEmptyValue` error since empty string is a valid value for this property. So enum type is not an option.